### PR TITLE
decouple `helm` deployer `chartPath` into `chartPath` and `remoteChart`

### PR DESCRIPTION
**Pull request by gsquared94**
Fixes: #5459  

---

**Description**

Currently the `helm` deployer's `chartPath` property can be used to specify the path to a local Helm chart (packaged or unpacked) or a Helm chart reference (like `stable/nginx`) or even a URL (like `https://example.com/charts/nginx-1.2.3.tgz`). It uses the additional `remote` property to hint skaffold about this.

This PR replaces `chartPath` and `remote` properties with explicit `chartPath` and `remoteChart` properties.

---

**User facing changes (remove if N/A)**

**Before**
```yaml
deploy:
  helm:
    releases:
    - name: foo1
      chartPath: ./path/to/chart
    - name: foo2
      chartPath: foo2/bar
      remote: true
```

**After**
```yaml
deploy:
  helm:
    releases:
    - name: foo1
      chartPath: ./path/to/chart
    - name: foo2
      remoteChart: foo2/bar
```

### CODE REVIEW

File: pkg/skaffold/deploy/helm/deploy.go [hunk 1]
```diff
@@ -253,7 +253,13 @@ func (h *Deployer) Render(ctx context.Context, out io.Writer, builds []build.Art
 	renderedManifests := new(bytes.Buffer)
 
 	for _, r := range h.Releases {
-		args := []string{"template", r.ChartPath}
+		var chartPath string
+		if r.LocalChartPath != "" {
+			chartPath = r.LocalChartPath
+		} else {
+			chartPath = r.RemoteChartPath
+		}
```
Comment (tejal29) [inline pkg/skaffold/deploy/helm/deploy.go:261]: How about reducing this or adding a utility function which 
```suggestion
		chartPath := r.RemoteChartPath
		if r.LocalChartPath != "" {
			chartPath = r.LocalChartPath
		}
```

e.g. 
```
func getChartPath(h latest.HelmRelease) string {
   if h.LocalChartPath != "" {
      return h.LocalChartPath
   } 
  return r.RemoteChartPath
}

```

**Review by tejal29 — COMMENTED** [review]
just a nit

File: pkg/skaffold/deploy/helm/deploy.go (hunk 1 above)
Comment (gsquared94) [reply pkg/skaffold/deploy/helm/deploy.go]: done

**Comment by MarlonGamez** [comment]
I have a question about this since https://github.com/GoogleContainerTools/skaffold/pull/5410 just got merged. Would it be possible to keep just the `chartPath` field, remove the `remote` field, and basically toggle on/off the remote functionality based on whether or not the user sets the `repo` field (introduced in #5410)?

**Comment by gsquared94** [comment]
> I have a question about this since #5410 just got merged. Would it be possible to keep just the `chartPath` field, remove the `remote` field, and basically toggle on/off the remote functionality based on whether or not the user sets the `repo` field (introduced in #5410)?

I think that approach won't work because the `chartPath` can be remote without specifying `--repo` if the user has already manually pulled it. Also it can be an absolute url in which case the `--repo` is not required as it is part of that url.

**Review by briandealwis — COMMENTED** [review]
`remoteChartPath` isn't actually a path.  Could we try the following:
- keep `chartPath` for local charts
- `remoteChartPath` &rarr; `chartRef` or `remoteChart`?

**Review by MarlonGamez — APPROVED** [review]
LGTM. Do we have anything in `examples/` that uses remote helm charts? If not I think that would be a good follow up for this

**Comment by gsquared94** [comment]
> LGTM. Do we have anything in `examples/` that uses remote helm charts? If not I think that would be a good follow up for this

Good idea. Added #5517

**Comment by zestrells** [comment]
Hello @gsquared94 , It looks like when we I try and use the remoteChart variable, it does not work. 
`deploy:
  helm:
    releases:
      - name: mysql
        namespace: data
        remoteChart: https://gitlab.com/helm-charts/devops/simple-template
        valuesFiles: [ helm/mysql-values.yaml ]
`

**Comment by briandealwis** [comment]
@zestrells try `skaffold fix`: it provides more detail on the issue.  I filed #5659 to improve this error message.

**Comment by briandealwis** [comment]
Oh and more specifically, that error often indicates that you need to update the version of your skaffold.yaml

**Pull request merged**